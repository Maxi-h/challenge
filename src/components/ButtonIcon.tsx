// Vendor
import React from 'react';
import {
  TouchableOpacity,
  ViewStyle,
  TextStyle,
  StyleProp,
  Image,
} from 'react-native';

// Colors
import colors from '../styles/colors';

// Types/Interfaces
type Variants = 'default' | 'inverted' | 'disabled';

interface Props {
  iconSize?: number;
  variant?: Variants;
  disabled?: boolean;
  path: string;
  style?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  color?: string;
  backgroundColor?: string;
  onTap?: any;
}

// Default props
const defaultProps: Partial<Props> = {
  variant: 'default',
  disabled: false,
  path: '',
  style: {},
  labelStyle: {},
  onTap: undefined,
  color: colors.white,
  iconSize: 30,
  backgroundColor: colors.main,
};

const ButtonIcon = ({
  disabled,
  path,
  style,
  iconSize,
  onTap,
  ...rest
}: Props) => (
  <TouchableOpacity {...rest} disabled={disabled} style={style} onPress={onTap}>
    {Boolean(path) && (
      <Image
        style={{width: iconSize, height: iconSize}}
        resizeMode="cover"
        source={require('../assets/images/menu.png')}
      />
    )}
  </TouchableOpacity>
);

ButtonIcon.defaultProps = defaultProps;

export default ButtonIcon;
