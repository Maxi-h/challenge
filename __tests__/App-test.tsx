/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import ButtonIcon from '../src/components/ButtonIcon';

describe('ButtonIcon Component', () => {
  it('has path', () => {
    const component = renderer.create(<ButtonIcon />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // eslint-disable-next-line jest/no-identical-title
  it('has path', () => {
    const component = renderer.create(
      <ButtonIcon path={'../assets/images/menu.png'} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
