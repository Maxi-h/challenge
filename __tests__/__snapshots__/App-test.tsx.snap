// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon Component has path 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
/>
`;

exports[`ButtonIcon Component has path 2`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/images/menu.png",
      }
    }
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  />
</View>
`;
